### PR TITLE
Refactor WebSocket to support SSL (wss)

### PR DESCRIPTION
**Pull request by dumganhar**
1. Supports wss protocol
2. Uses only one libwebsockets context, multiple wsi instances to avoid multi-thread issues
3. Uses lws_create_vhost to create multiple virtual host, different protocols works

This PR depends on https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin/pull/263

Discussion: http://discuss.cocos2d-x.org/t/websocket-supports-ssl-connections/34611

### CODE REVIEW

**Comment by dumganhar** [comment]
@minggo, please review.

File: build/cocos2d_libs.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -17451,7 +17447,7 @@
 				OTHER_LDFLAGS = "-llibsql3";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SDKROOT = appletvos9.1;
-				USER_HEADER_SEARCH_PATHS = "$(inherited) $(SRCROOT)/../external/freetype2/include/ios $(SRCROOT)/../external/curl/include/ios $(SRCROOT)/../external/webp/include/ios $(SRCROOT)/../external/tiff/include/ios $(SRCROOT)/../external/jpeg/include/ios $(SRCROOT)/../external/png/include/ios $(SRCROOT)/../external/websockets/include/ios $(SRCROOT)/../external/freetype2/include/ios/freetype2";
+				USER_HEADER_SEARCH_PATHS = "$(inherited) $(SRCROOT)/../external/freetype2/include/ios $(SRCROOT)/../external/curl/include/ios $(SRCROOT)/../external/webp/include/ios $(SRCROOT)/../external/tiff/include/ios $(SRCROOT)/../external/jpeg/include/ios $(SRCROOT)/../external/png/include/ios $(SRCROOT)/../external/websockets/include/ios $(SRCROOT)/../external/freetype2/include/ios/freetype2 $(SRCROOT)/../external/openssl/include/ios";
 			};
```
Comment (minggo) [inline build/cocos2d_libs.xcodeproj/project.pbxproj:17451]: Should change template too?

Comment (dumganhar) [reply build/cocos2d_libs.xcodeproj/project.pbxproj:17451]: Yes, we should, I have forgotten.
Does Jenkins compile template?

Comment (minggo) [reply build/cocos2d_libs.xcodeproj/project.pbxproj:17451]: no, jenkins doesn't compile template to reduce compiling time

Comment (dumganhar) [reply build/cocos2d_libs.xcodeproj/project.pbxproj:17451]: @minggo, I tested iOS template, it worked fine without any modifications.

**Comment by dhb118** [comment]
compile fail at cocos2d_tests.xcodeproj target 'cpp-tests Mac'
xcode version:8.2 beta
macOS SDK:10.12
e8dd3a13-61e3-4fd6-b685-e998aa6a8d36

**Comment by dumganhar** [comment]
@dhb118, I guess you're using git. It's a bug of `download-deps.py`, I fixed it at https://github.com/cocos2d/cocos2d-x/pull/17247 .
You just need to pull the latest code, and execute `download-deps.py` again.

**Comment by yuanyouli** [comment]
it will crash when close WebClient at IOS  same as the cpp-tests demo

**Pull request merged**